android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/Worker.kt

```diff
@@ -22,6 +22,16 @@ package com.uber.rib.core
  * other lifecycles we're interested in.
  */
 public interface Worker {
+
+  /**
+   * Specifies in which ThreadingType the WorkerBinder will bound this Worker.
+   *
+   * Whe not overriden, will use the thread where [WorkerBinder.bind] call happens
+   */
+  @JvmDefault
+  public val workerBinderThreadingType: WorkerBinderThreadingType
+    get() = WorkerBinderThreadingType.CALLER_THREAD
```

Comment:
Decided against defaulting to `BACKGROUND_COMPUTATION` for retro compatibility?

Reply:
That's right. Given this will be running on existing usages we need to guarantee the current threading model is honored (otherwise we may face unwanted concurrent/not on ui thread crashes)

With the added monitoring will have an option now to detect workers on UI thread and move those to background